CurvesAlgo::resamplePrimitiveVariables : Support periodic curves

danieldresser-ie opened the pull request:
Not sure how much periodic curves are ever actually used, but as long as our curves type supports them, we probably shouldn't output bogus data for them.

johnhaddon commented on src/IECoreScene/CurvesAlgo.cpp:
```diff
@@ -200,7 +200,17 @@ struct  CurvesUniformToVarying
 		{
 			Canceller::check( m_canceller );
-			for( size_t j = 0; j < m_curves->numSegments( i ) + 1; ++j )
+			size_t varyingSize;
+			if( m_curves->periodic() )
+			{
+				varyingSize = m_curves->numSegments( i );
+			}
+			else
+			{
+				varyingSize = m_curves->numSegments( i ) + 1;
+			}
+
+			for( size_t j = 0; j < varyingSize; ++j )
```
Could be a one liner I think :

`const size_t varyingSize = m_curves->variableSize( PrimitiveVariable::Varying, i )`.

johnhaddon replied:
Same applies to the other changes below...

johnhaddon commented on test/IECoreScene/CurvesAlgoTest.py:
```diff
@@ -470,6 +470,38 @@ def testBSplineCurvesIndexedFaceVaryingToVarying( self ) :
+
+		p = IECoreScene.PrimitiveVariable( IECoreScene.PrimitiveVariable.Interpolation.Vertex, IECore.FloatVectorData( [ i for i in range( curves.variableSize( IECoreScene.PrimitiveVariable.Interpolation.Vertex ) ) ] ) )
+
+		# Work around weird limitation that for resampling Varying -> Vertex the primvar must be stored on curves
+		curves["dummy"] = p
```
I briefly tried to figure out why this might be necessary, but failed. Could you educate me?

johnhaddon reviewed (commented):
Thanks Daniel! I think it's probably worth taking the one-liner approach (see comment inline)?

danieldresser-ie commented:
I've switched to using variableSize, and added some more info to the comment ( looks like there isn't any good reason why CurvesAlgo.resamplePrimitiveVariable is doing weird things, but that's a fix for another day ).

Should be good to merge.
